cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_sqrt_iswap.py

```diff
@@ -32,6 +33,201 @@
     import cirq
 
 
+def parameterized_2q_op_to_sqrt_iswap_operations(
+    op: 'cirq.Operation', *, use_sqrt_iswap_inv: bool = False
+) -> protocols.decompose_protocol.DecomposeResult:
+    """Tries to decompose a parameterized 2q operation into √iSWAP's + parameterized 1q rotations.
+
+    Currently only supports decomposing the following gates:
+        a) `cirq.CZPowGate`
+        b) `cirq.SwapPowGate`
+        c) `cirq.ISwapPowGate`
+        d) `cirq.FSimGate`
+
+    Args:
+        op: Parameterized two qubit operation to be decomposed into sqrt-iswaps.
+        use_sqrt_iswap_inv: If True, `cirq.SQRT_ISWAP_INV` is used as the target 2q gate, instead
+            of `cirq.SQRT_ISWAP`.
+
+    Returns:
+        A parameterized `cirq.OP_TREE` implementing `op` using only `cirq.SQRT_ISWAP`
+        (or `cirq.SQRT_ISWAP_INV`) and parameterized single qubit rotations OR
+        None or NotImplemented if decomposition of `op` is not known.
```

Comment:
Is it an op_tree or a DecomposeResult ? (from fxn return type hint)

Reply:
`DecomposeResult` is `Union[cirq.OP_TREE, None, NotImplementedType]` -- the docstring covers all 3 cases.